crates/swc_ecma_parser/src/parser/object.rs

```diff
@@ -51,6 +51,23 @@ impl<I: Tokens> Parser<I> {
         .parse_with(|p| {
             let start = cur_pos!(p);
 
+            // The lexer marks keywords with escapes as errors, but unlike in
+            // the general case, it's OK here as a property name. We turn the
+            // error back into a token
+            if let Some(err_tok @ Token::Error(e)) = p.input.cur() {
+                if let SyntaxError::EscapeInReservedWord { word } = e.kind() {
+                    let word = word.clone();
+                    let err_tok = err_tok.clone();
+                    let check = p.input.bump();
+                    debug_assert_eq!(err_tok, check);
+                    eprintln!(
+                        "  :: parser.object.parse_prop_name: recovered error {:?} -> {:?}",
+                        err_tok, word
+                    );
+                    p.input.store(Token::Word(word));
+                }
+            }
```

Comment:
This entire logic could be refactored into a method in `Buffer`, something like `p.input.unget` (reusing the from the C; or maybe "unread" like in Go)